src/csgnode.cc

```diff
@@ -159,19 +150,62 @@ std::string CSGLeaf::dump()
 	return this->label;
 }
 
-std::string CSGOperation::dump()
+// Recursive traversal can cause stack overflow with very large loops of child nodes,
+// so tree is traverse iteratively, managing our own stack.
+std::string CSGOperation::dump() 
```

Comment:
Optional: It feels like this method should be const

Reply:
Agreed, I tried making it const early on, before I had finalized what exact types I was using on stack etc, but ran into some snags.  Just gave it another try and got it working this time.